src/object.rs

```diff
@@ -583,6 +583,42 @@ glib_object_wrapper! {
     Object, GObject, GObjectClass, @get_type gobject_ffi::g_object_get_type()
 }
 
+impl Object {
+    pub fn new(type_: Type, properties: &[(&str, &ToValue)]) -> Result<Object, BoolError> {
+        use std::ffi::CString;
+
+        if !type_.is_a(&Object::static_type()) {
+            return Err(BoolError("Can't instantiate non-GObject objects"));
+        }
+
+        let params = properties.iter()
+                               .map(|&(name, value)|
+                                    (CString::new(name).unwrap(), value.to_value()))
+                               .collect::<Vec<_>>();
```

Comment:
We could also check for all properties if they actually exist and are of the given value, and assert otherwise or return an error.